plugins/PostProcessingPlugin/scripts/MultiExtColorMix.py

```diff
@@ -0,0 +1,623 @@
+            Message(text = "Color Mixer: {}".format(textstring)).show()
+            data[0] += ";  Multi-Extruder Color Mixer did not run because - Start Total Percent = " + str(total_start_percent) + " EndTotalPercent = " + str(total_end_percent) + " Both sums must equal 100\n"
+            return data
+        
+        # Put together the Reset String-----------------------------------------------------------
+        if extruder_count == 2:
+            M163_reset = "M163 P0.50 S0" + "\nM163 P0.50 S1\n"
+        if extruder_count == 3:
+            M163_reset = "M163 P0.33 S0" + "\nM163 P0.34 S1\n" + "M163 P0.33 S2\n"
+        M163_reset += "M164 S" + str(M164_extruder) + "\n"
+            
+        # Put together the Initial Mix String----------------------------------------------------------    
+        if T0_include:
+            M163_T0 = "\nM163 P" + str(T0_mix_start/100) + " S0"
+        else:
+            M163_T0 = ""
+        if T1_include:
+            M163_T1 = "\nM163 P" + str(T1_mix_start/100) + " S1"
+        else:
+            M163_T1 = ""
+        if T2_include:
+            M163_T2 = "\nM163 P" + str(T2_mix_start/100) + " S2"
+        else:
+            M163_T2 = ""
+        M164str = M163_T0 + M163_T1 + M163_T2 + "\nM164 S" + str(M164_extruder) + "\nT" + str(M164_extruder)
+        
+        # If purge is selected--------------------------------------------------------
+        initial_purge = ""
+        start_purge = ""
+        final_string = ""
+        if park_head:
+            initial_purge = ParkAndPurge.park_script(0, data, park_x, park_y, purge_amt)[1]
+        if park_head and M164_extruder != resume_extruder:
+            start_purge = ParkAndPurge.park_script(start_layer, data, park_x, park_y, purge_amt)[1]
+        if park_head and self.getSettingValueByKey("end_layer") != -1:
+            final_string = ParkAndPurge.park_script(end_layer, data, park_x, park_y, purge_amt)[0]
+        
+        # Check to see if a reset is already present and if so then move on-------------------------------
+        reset_present = False
+        start_up_sect = data[1]
+        lines = start_up_sect.split("\n")
+        lines.reverse()
+        for z_index in range(0,7,1):          
+            if lines[z_index].startswith("M164") or "Purge" in lines[z_index]:
+                reset_present = True
+                break
+        if not reset_present:
+            INIT_str = M163_T0 + M163_T1 + M163_T2 + "\nM164 S" + str(M164_extruder) + "\nT" + str(init_extruder)
+            if start_layer > 0: INIT_str += initial_purge
+            lines = data[1].split("\n")
+            lines.insert(len(lines)-2, INIT_str[1:])
+            data[1] = "\n".join(lines)
+        
+        pre_ret = False
+        post_prime = False
+        for index, layer in enumerate(data):
+        # Constant-------------------------------------------------------------------------------
+            if mix_style == "constant":
```

Comment:
I would change two things here,
1.  instead of using `if mix_style == "A" ... if mix_style == "B"` i would use the [match pattern ](https://peps.python.org/pep-0622/)
2. The code is a bit dependent on a lot of properties so this might be harder but since it is already such a long function i would like for the different mix strategies to be separate calls

```suggestion
            match mix_style:
               case "constant":
                    self.processConstant(...)
                case "gradient":
                    self.processGradient(...)
```

Reply:
Done.